### DISCUSSION TITLE
Android 13 Support in the Flutter Firebase Messaging Package

### DISCUSSION THREAD

**Discussion by assemblethis** [firebase/flutterfire#9130] [Feature request]
Hello Firebase Messaging team,

Thanks for your work in creating a Flutter package for Firebase messaging. I'm wondering if the firebase_messaging flutter package will be targeting Android 13 soon? I'm currently using this package on iOS to request the Notification permission, but similar functionality is not supported in the Android plugin yet. I'm considering using the permission_handler flutter package which already supports Android 13 Notification permissions in the recently released v10.0.0. Is that the preferred direction for handling permissions using Flutter? 

I'm currently using the following to request the Notification permission on iOS: 
```
FirebaseMessaging.instance.requestPermission(alert: true, announcement: false, badge: true, carPlay: false, criticalAlert: false, provisional: false, sound: true);
```

Will this method be updated to support the Notification permission that's coming in Android 13? If not, are there any recommendations on how to handle the Android 13 Notification permission in Flutter?

Thank you!

**Comment by assemblethis** [comment]
[Some discussion here](https://github.com/firebase/flutterfire/issues/9279)

**Comment by guulp** [comment]
Also, see #8720
